src/sources/http_scrape/scrape.rs

```diff
@@ -0,0 +1,549 @@
+            }
+            assert_eq!(got, expected);
+        }
+    }
+}
+
+#[cfg(all(test, feature = "http-scrape-integration-tests"))]
+mod integration_tests {
+    use tokio::time::Duration;
+
+    use super::*;
+    use crate::test_util::components::{run_and_assert_source_compliance, HTTP_PULL_SOURCE_TAGS};
+
+    async fn run_test(config: HttpScrapeConfig) -> Vec<Event> {
+        let events = run_and_assert_source_compliance(
+            config,
+            Duration::from_secs(1),
+            &HTTP_PULL_SOURCE_TAGS,
+        )
+        .await;
+        assert!(!events.is_empty());
+        events
+    }
+
+    #[tokio::test]
+    async fn scraped_logs_bytes() {
+        run_test(HttpScrapeConfig {
+            endpoint: "http://dufs:5000/logs/bytes".to_string(),
+            scrape_interval_secs: 1,
+            query: None,
+            decoding: DeserializerConfig::Bytes,
+            framing: default_framing_message_based(),
+            headers: None,
+            auth: None,
+            tls: None,
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn scraped_logs_json() {
+        let events = run_test(HttpScrapeConfig {
+            endpoint: "http://dufs:5000/logs/json.json".to_string(),
+            scrape_interval_secs: 1,
+            query: None,
+            decoding: DeserializerConfig::Json,
+            framing: default_framing_message_based(),
+            headers: None,
+            auth: None,
+            tls: None,
+        })
+        .await;
+        let log = events[0].as_log();
+        assert_eq!(log[log_schema().source_type_key()], NAME.into());
+    }
+
+    #[tokio::test]
+    async fn scraped_metrics_native_json() {
+        let events = run_test(HttpScrapeConfig {
+            endpoint: "http://dufs:5000/metrics/native.json".to_string(),
+            scrape_interval_secs: 1,
+            query: None,
+            decoding: DeserializerConfig::NativeJson,
+            framing: default_framing_message_based(),
+            headers: None,
+            auth: None,
+            tls: None,
+        })
+        .await;
+
+        let metric = events[0].as_metric();
+        assert_eq!(
+            metric.tags().unwrap()[log_schema().source_type_key()],
+            NAME.to_string()
+        );
+    }
+
+    #[tokio::test]
+    async fn scraped_trace_native_json() {
+        let events = run_test(HttpScrapeConfig {
+            endpoint: "http://dufs:5000/traces/native.json".to_string(),
+            scrape_interval_secs: 1,
+            query: None,
+            decoding: DeserializerConfig::NativeJson,
+            framing: default_framing_message_based(),
+            headers: None,
+            auth: None,
+            tls: None,
+        })
+        .await;
+
+        let trace = events[0].as_trace();
+        assert_eq!(trace.as_map()[log_schema().source_type_key()], NAME.into());
+    }
+
+    #[tokio::test]
+    async fn unauthorized() {
+        // TODO how to assert failure
```

Comment:
`#[should_panic]`

Reply:
The problem isn't the asserting I guess- its how to actually surface the failure.
I'm currently working on this at the moment- trying to get the error from the source's future (working on that for the `invalid_endpoint` test, but the same needs to happen here.

Reply:
Ah, gotcha misunderstood

Reply:
Nah, I said `assert` lol, can't you read my mind yet ? 😛